File: nodejs-examples/test-offline-nemo-canary.js
// Copyright (c)  2025  Xiaomi Corporation (authors: Fangjun Kuang)
//
const fs = require('fs');
const {Readable} = require('stream');
const wav = require('wav');

const sherpa_onnx = require('sherpa-onnx');

function createOfflineRecognizer() {
  let config = {
    modelConfig: {
      canary: {
        encoder:
            './sherpa-onnx-nemo-canary-180m-flash-en-es-de-fr-int8/encoder.int8.onnx',
        decoder:
            './sherpa-onnx-nemo-canary-180m-flash-en-es-de-fr-int8/decoder.int8.onnx',
        srcLang: 'en',
        dstLang: 'en',
        usePnc: 1,
      },
      tokens:
          './sherpa-onnx-nemo-canary-180m-flash-en-es-de-fr-int8/tokens.txt',
    }
  };

  return sherpa_onnx.createOfflineRecognizer(config);
}

const recognizer = createOfflineRecognizer();
let stream = recognizer.createStream();

const waveFilename =
    './sherpa-onnx-nemo-canary-180m-flash-en-es-de-fr-int8/test_wavs/en.wav';
const wave = sherpa_onnx.readWave(waveFilename);
stream.acceptWaveform(wave.sampleRate, wave.samples);

recognizer.decode(stream);
let text = recognizer.getResult(stream).text;
console.log(`text in English: ${text}`);

stream.free();

// now output German text
recognizer.config.modelConfig.canary.tgt_lang = 'de';
recognizer.setConfig(recognizer.config);

stream = recognizer.createStream();
stream.acceptWaveform(wave.sampleRate, wave.samples);
recognizer.decode(stream);
text = recognizer.getResult(stream).text;

console.log(`text in German: ${text}`);

stream.free();
recognizer.free();

File: kotlin-api-examples/test_offline_nemo_canary.kt
package com.k2fsa.sherpa.onnx

fun main() {
  val recognizer = createOfflineRecognizer()
  val waveFilename = "./sherpa-onnx-nemo-canary-180m-flash-en-es-de-fr-int8/test_wavs/en.wav"

  val waveData = WaveReader.readWaveFromFile(
      filename = waveFilename,
  )

  var stream = recognizer.createStream()
  stream.acceptWaveform(waveData.samples, sampleRate=waveData.sampleRate)
  recognizer.decode(stream)

  var result = recognizer.getResult(stream)
  println("English: $result")

  stream.release()

  // now output text in German
  val config = recognizer.config.copy(modelConfig=recognizer.config.modelConfig.copy(
    canary=recognizer.config.modelConfig.canary.copy(
      tgtLang="de"
    )
  ))
  recognizer.setConfig(config)

  stream = recognizer.createStream()
  stream.acceptWaveform(samples, sampleRate=sampleRate)
  recognizer.decode(stream)

  result = recognizer.getResult(stream)
  println("German: $result")

  stream.release()
  recognizer.release()
}


fun createOfflineRecognizer(): OfflineRecognizer {
  val config = OfflineRecognizerConfig(
      modelConfig = getOfflineModelConfig(type = 32)!!,
  )

  return OfflineRecognizer(config = config)
}
